src/wp-includes/post.php

```diff
@@ -69,6 +69,11 @@ function create_initial_post_types() {
 		)
 	);
 
+	// Enhance page editor for block themes by rendering template and content blocks.
+	if ( wp_is_block_theme() ) {
```

Comment:
```suggestion
	if ( wp_is_block_theme() && current_theme_supports( 'block-templates' ) ) {
```

I think I'll need to harden this condition by adding a `current_theme_supports` check. While it should be an edge case for a block theme not to support `block-templates`, it's still technically doable. See: https://github.com/WordPress/gutenberg/issues/67875#issuecomment-2655718449.